src/publish/s3.js

```diff
@@ -3,9 +3,22 @@ const winston = require("winston-color")
 const zlib = require("zlib")
 const config = require("../config")
 
-const S3 = new AWS.S3()
+let conf = {}
 
+// This is the case where using custom s3 api-like services like minio.
+if (config.aws.endpoint) {
+  conf = {
+    accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+    secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+    endpoint: (config.aws.endpoint ? config.aws.endpoint : ''),
+    s3ForcePathStyle: 'true',
+    signatureVersion: 'v4'
+  }
```

Comment:
Does it hurt to pass in `s3ForcePathStyle: 'true'` and `signatureVersion: 'v4'` as arguments to `conf` even if not using a custom endpoint?

I think it'd be nice if the conditional logic here was removed, and we instead just had `config.aws.endpoint` default to `null` (or whatever causes `S3` to use its built-in defaults).